pupil_src/shared_modules/video_overlay/utils/image_manipulation.py

```diff
@@ -6,26 +6,32 @@
 
 class ImageManipulator(metaclass=abc.ABCMeta):
     @abc.abstractmethod
-    def apply_to(self, image, parameter):
+    def apply_to(self, image, parameter, **kwargs):
         raise NotImplementedError
 
 
 class ScaleTransform(ImageManipulator):
-    def apply_to(self, image, parameter):
+    def apply_to(self, image, parameter, **kwargs):
         """parameter: scale factor as float"""
         return cv2.resize(image, (0, 0), fx=parameter, fy=parameter)
 
 
 class HorizontalFlip(ImageManipulator):
-    def apply_to(self, image, parameter):
+    def apply_to(self, image, parameter, *, is_fake_frame, **kwargs):
```

Comment:
@papr why does `ImageManipulator` take an `image` as an argument and not a `frame`? if it did, the `is_fake_frame` argument wouldn't be needed, since all the relevant info to perform the manipulation would be provided in `frame`.

Reply:
@papr I like the idea of passing the whole frame object to the image manipulators.
Anything speaking against this? Should I change it that way?

Reply:
@romanroibu @pfaion The initial idea was to pass the minimum required data object (the image buffer). Since the requirements changed, one could argue the next smallest data object is the frame.

@pfaion feel free to make that change.

Reply:
@papr @romanroibu 
Turns out this is not so easy currently, as the image property in our frame class is read-only, as it is inferred from the underlying av_frame.

More specifically: In order for this to work, we'd have to change the ImageManipulators to receive and return frames. In `apply_to()` we'd have to modify the image of the frame. But the file_backend frames don't really have an image, but only an internal av_frame, that get's converted into a numpy-array when accessing the `Frame.img` property. Thus we can't consistently modify the image of a frame.

Maybe we'd be able to actually remove the dependency on the av_frame internally, but I'd argue to not do this now. This might become clearer, once we implement a clear pupil frame class that we use across all backends. We will have to touch this code again then anyways.

If we leave it like this, I'd just add a NOTE to change this when we refactor the frame class.
Thoughts?

Reply:
Leave as it is for now.